examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -773,9 +773,11 @@ void * wfx_rsi_alloc_pkt(uint16_t data_length)
 
     status = sl_si91x_allocate_command_buffer(&buffer, (void **) &packet, sizeof(sl_si91x_packet_t) + data_length,
                                               SL_WIFI_ALLOCATE_COMMAND_BUFFER_WAIT_TIME_MS);
-    if (packet == NULL)
+    if ((packet == NULL) || (status != SL_STATUS_OK))
     {
         return SL_STATUS_ALLOCATION_FAILED;
+        SILABS_LOG("packet allocation failed: %d", status);
```

Comment:
you have a return just on top, new code under it won't run.

Returning a void * that is either a sl_si91x_packet_t * or a sl_status_t is really not ideal.
packet is also a local variable and will be on the stack. Once out of the function, the stack can be reused and data be modified...

This function should be changed to something like this
`sl_status_t wfx_rsi_alloc_pkt(uint16_t data_length, sl_si91x_packet_t * packet)`
and packet be create by the caller

Reply:
Removed wfx_rsi_alloc_pkt() function since it is not being used anywhere.